contracts/tokens/STFactory.sol

```diff
@@ -31,24 +65,94 @@ contract STFactory is ISTFactory {
         bool _divisible,
         address _treasuryWallet,
         address _polymathRegistry
-    ) 
-        external 
-        returns(address) 
+    )
+        external
+        returns(address)
     {
-        SecurityToken newSecurityToken = new SecurityToken(
+        address securityToken = _deploy(
             _name,
             _symbol,
             _decimals,
-            _divisible ? 1 : uint256(10) ** _decimals,
             _tokenDetails,
-            _polymathRegistry,
-            stDelegate
+            _divisible,
+            _polymathRegistry
         );
         //NB When dataStore is generated, the security token address is automatically set via the constructor in DataStoreProxy.
-        newSecurityToken.changeDataStore(dataStoreFactory.generateDataStore(address(newSecurityToken)));
-        newSecurityToken.changeTreasuryWallet(_treasuryWallet);
-        newSecurityToken.addModule(transferManagerFactory, "", 0, 0);
-        newSecurityToken.transferOwnership(_issuer);
-        return address(newSecurityToken);
+        ISecurityToken(securityToken).changeDataStore(dataStoreFactory.generateDataStore(securityToken));
+        ISecurityToken(securityToken).changeTreasuryWallet(_treasuryWallet);
+        ISecurityToken(securityToken).addModule(transferManagerFactory, "", 0, 0, false);
+        IOwnable(securityToken).transferOwnership(_issuer);
+        return securityToken;
     }
+
+    function _deploy(
+        string memory _name,
+        string memory _symbol,
+        uint8 _decimals,
+        string memory _tokenDetails,
+        bool _divisible,
+        address _polymathRegistry
+    ) internal returns(address) {
+        // Creates proxy contract and sets some initial storage
+        SecurityTokenProxy proxy = new SecurityTokenProxy(
+            _name,
+            _symbol,
+            _decimals,
+            _divisible ? 1 : uint256(10) ** _decimals,
+            _tokenDetails,
+            _polymathRegistry
+        );
+        // Sets logic contract
+        proxy.upgradeTo(logicContracts[latestUpgrade].version, logicContracts[latestUpgrade].logicContract);
```

Comment:
STF should deploy genesis version of ST rather than the latest version.

Reply:
Don't agree - otherwise newly generated tokens will be on old versions. I think everyone should be forced to use the latest minor / patch version of a major version. Major version can be selected through the STR.